llvm/include/llvm/IR/IntrinsicsNVVM.td

```diff
@@ -4710,4 +4710,10 @@ def int_nvvm_is_explicit_cluster
               [IntrNoMem, IntrSpeculatable, NoUndef<RetIndex>],
               "llvm.nvvm.is_explicit_cluster">;
 
+// Setmaxnreg intrinsic
+def int_nvvm_setmaxnreg_sync_aligned_u32
+  : DefaultAttrsIntrinsic<[], [llvm_i32_ty, llvm_i32_ty],
+              [IntrConvergent, IntrNoMem, IntrHasSideEffects, ImmArg<ArgIndex<1>>],
```

Comment:
Why is this intrinsic convergent?

Reply:
The PTX instruction has the '.aligned' semantics i.e. all threads must execute the same setmaxnreg instruction.

Quoting from the PTX spec:
--
"The mandatory .aligned qualifier indicates that all threads in the warpgroup must execute the same setmaxnreg instruction. In conditionally executed code, setmaxnreg instruction should only be used if it is known that all threads in warpgroup evaluate the condition identically, otherwise the behavior is undefined."

Reply:
Does "IntrConvergent" means convergence within a warpgroup? This instruction helps establish smaller register sizes for first warpgroup and larger register sizes for second warpgroup. At least this is how we use for warp specilized GEMM kernels

Reply:
Yes,